app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java

```diff
@@ -18,7 +18,7 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    public State state;
+    public State state = State.LOGIN;
```

Comment:
why is this outside of the onStart method of the main activity?

Reply:
Because it's a class variable that the fragments need access too